# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1489596187 14400
#      Wed Mar 15 12:43:07 2017 -0400
# Node ID eb19c1545fb85397efb0f36fb288f2e4f03bda99
# Parent  df0fb2cb820be2efecc03e47aaeae4554b79516e
display fntests.log at end of make check if V=1

* test/module.mk (run-octave-tests): New macro.  Optionally display
fntests.log file.
(check-local, check-jit): Use new macro.

diff --git a/test/module.mk b/test/module.mk
--- a/test/module.mk
+++ b/test/module.mk
@@ -60,22 +60,36 @@ include test/fcn-handle-derived-resoluti
 include test/nest/module.mk
 include test/publish/module.mk
 
 ALL_LOCAL_TARGETS += test/.gdbinit
 
 test/.gdbinit: etc/gdbinit
 	@$(gdbinit_install_rule)
 
+define run-octave-tests
+  cd test && $(SHELL) ../run-octave $(RUN_OCTAVE_OPTIONS) $(1) --norc --silent --no-history $(abs_top_srcdir)/test/fntests.m $(abs_top_srcdir)/test
+  if $(AM_V_P); then \
+    echo ""; \
+    if [ -f test/fntests.log ]; then \
+      echo "Contents of test/fntests.log:"; \
+      echo ""; \
+      cat test/fntests.log; \
+    else \
+      echo "test/fntests.log is missing!"; \
+    fi; \
+  fi
+endef
+
 check-local: $(GENERATED_TEST_FILES) | $(OCTAVE_INTERPRETER_TARGETS) test/$(octave_dirstamp)
-	cd test && $(SHELL) ../run-octave $(RUN_OCTAVE_OPTIONS) --norc --silent --no-history $(abs_top_srcdir)/test/fntests.m $(abs_top_srcdir)/test
+	$(call run-octave-tests)
 
 if AMCOND_HAVE_LLVM
 check-jit: $(GENERATED_TEST_FILES) | $(OCTAVE_INTERPRETER_TARGETS) test/$(octave_dirstamp)
-	cd test && $(SHELL) ../run-octave $(RUN_OCTAVE_OPTIONS) --jit-compiler --norc --silent --no-history $(abs_top_srcdir)/test/fntests.m $(abs_top_srcdir)/test
+	$(call run-octave-tests,--jit-compiler)
 endif
 
 COVERAGE_DIR = test/coverage
 COVERAGE_INFO = $(COVERAGE_DIR)/$(PACKAGE).info
 
 ## FIXME: To get something useful out of 'make coverage', you should use gcc
 ## and configure with compiler flags set to '-g --coverage'.  Adding the
 ## --coverage option to either WARN_CXXFLAGS or XTRA_CXXFLAGS resulted in
